File: iosApp/iosApp/ComponentViews/StopCard/LoadingStopCard.swift
//
//  LoadingStopCard.swift
//  iosApp
//
//  Created by Melody Horn on 6/25/26.
//  Copyright © 2026 MBTA. All rights reserved.
//

import Shared
import SwiftUI

struct LoadingStopCard: View {
    @ObserveInjection var inject
    var body: some View {
        let placeholderRouteData = LoadingPlaceholders.shared.nearbyRoute()
        let placeholderStopData = StopCardData.companion.fromRouteCardData(
            routeCardData: [placeholderRouteData],
            sortByDistanceFrom: nil
        )
        if let stopData = placeholderStopData.first {
            StopCard(
                cardData: stopData,
                global: nil,
                now: EasternTimeInstant.now(),
                isFavorite: { _ in false },
                pushNavEntry: { _ in }
            )
            .enableInjection()
        }
    }
}

File: androidApp/src/main/java/com/mbta/tid/mbta_app/android/component/stopCard/LoadingStopCard.kt
package com.mbta.tid.mbta_app.android.component.stopCard

import androidx.compose.foundation.layout.Column
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.mbta.tid.mbta_app.android.util.modifiers.loading
import com.mbta.tid.mbta_app.model.LoadingPlaceholders
import com.mbta.tid.mbta_app.model.StopCardData
import com.mbta.tid.mbta_app.utils.EasternTimeInstant

@Composable
fun LoadingStopCard() {
    val placeholderRouteData = LoadingPlaceholders.nearbyRoute()
    val placeholderStopData =
        StopCardData.fromRouteCardData(listOf(placeholderRouteData), sortByDistanceFrom = null)
    Column(modifier = Modifier.loading(withShimmer = false)) {
        StopCard(
            placeholderStopData.first(),
            null,
            EasternTimeInstant.now(),
            { false },
            { _, _ -> },
        )
    }
}
